pallets/avn-anchor/src/lib.rs

```diff
@@ -6,62 +6,110 @@ extern crate alloc;
     use frame_system::pallet_prelude::*;
 
+    pub type ChainId = u32;
+
     #[pallet::config]
-    pub trait Config: frame_system::Config + pallet_avn::Config 
-    {
-        // type RuntimeEvent: From<Event<Self>>
-        //     + Into<<Self as frame_system::Config>::RuntimeEvent>
-        //     + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+    pub trait Config: frame_system::Config + pallet_avn::Config {
+        type RuntimeEvent: From<Event<Self>>
+            + Into<<Self as frame_system::Config>::RuntimeEvent>
+            + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
         type WeightInfo: WeightInfo;
     }
 
     #[pallet::pallet]
     pub struct Pallet<T>(_);
 
-    // #[pallet::event]
-    // #[pallet::generate_deposit(pub(super) fn deposit_event)]
-    // pub enum Event<T: Config> {}
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// A new chain handler was registered. [chain_id, handler_account_id]
+        ChainHandlerRegistered(ChainId, T::AccountId),
+        /// A chain handler was updated. [chain_id, new_handler_account_id]
+        ChainHandlerUpdated(ChainId, T::AccountId),
+    }
 
     #[pallet::error]
-    pub enum Error<T> {}
+    pub enum Error<T> {
+        HandlerAlreadyExists,
+        HandlerNotRegistered,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn chain_handlers)]
+    pub type ChainHandlers<T: Config> = StorageMap<_, Blake2_128Concat, ChainId, T::AccountId>;
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
         #[pallet::weight(<T as pallet::Config>::WeightInfo::register_chain_handler())]
         #[pallet::call_index(0)]
         pub fn register_chain_handler(
             origin: OriginFor<T>,
-            // handler:
+            chain_id: ChainId,
+            handler_account_id: T::AccountId,
         ) -> DispatchResult {
+            ensure_signed(origin)?;
+
+            ensure!(!ChainHandlers::<T>::contains_key(&chain_id), Error::<T>::HandlerAlreadyExists);
+
+            ChainHandlers::<T>::insert(chain_id, handler_account_id.clone());
+
+            Self::deposit_event(Event::ChainHandlerRegistered(
+                chain_id,
+                handler_account_id.clone(),
+            ));
+
             Ok(())
         }
 
-        #[pallet::weight( <T as pallet::Config>::WeightInfo::update_chain_handler())]
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::update_chain_handler())]
         #[pallet::call_index(1)]
         pub fn update_chain_handler(
             origin: OriginFor<T>,
-            // handler:
+            chain_id: ChainId,
+            new_handler_account_id: T::AccountId,
         ) -> DispatchResult {
+            ensure_signed(origin)?;
+
+            ChainHandlers::<T>::try_mutate(&chain_id, |maybe_handler| -> DispatchResult {
+                let handler = maybe_handler.as_mut().ok_or(Error::<T>::HandlerNotRegistered)?;
```

Comment:
I think if the data structure is chain_id -> handler if the chain id is not returning a handler then we should error with something that says that "chain_id is not registered" rather that the handler was not found.